essence-of-live-coding/test/Cell.hs

```diff
@@ -23,10 +25,49 @@ import LiveCoding
 
 import qualified Cell.Util
 import qualified Cell.Monad.Trans
+import Control.Exception (ErrorCall)
+import Test.Framework.Providers.HUnit
+import Control.Exception.Base
+import Test.HUnit
 
 test = testGroup "Cell"
   [ testProperty "steps produces outputs"
     $ \(inputs :: [Int]) -> inputs === fst (runIdentity $ steps (id :: Cell Identity Int Int) inputs)
+  , testProperty "traverse' works as expected for Lists"
+  $ \(inputs :: [[Int]]) -> fmap (fmap (+10)) inputs === fst (runIdentity $ steps (traverse' (arr (+10) :: Cell Identity Int Int)) inputs)
+  , testProperty "traverse' works as expected for Maybes"
+  $ \(inputs :: [Maybe Int]) -> fmap (fmap (+10)) inputs === fst (runIdentity $ steps (traverse' (arr (+10) :: Cell Identity Int Int)) inputs)
+  , testProperty "traverse' for [] is equal to resampleList"
+  $ \(inputs :: [[Int]]) -> fst (runIdentity $ steps (resampleList (arr (+10) :: Cell Identity Int Int)) inputs)
+    === fst (runIdentity $ steps (traverse' (arr (+10) :: Cell Identity Int Int)) inputs)
+  , testProperty "traverse' for Maybe is equal to resampleMaybe"
+  $ \(inputs :: [Maybe Int]) -> fst (runIdentity $ steps (resampleMaybe (arr (+10) :: Cell Identity Int Int)) inputs)
+    === fst (runIdentity $ steps (traverse' (arr (+10) :: Cell Identity Int Int)) inputs)
+  , testCase "traverse' by itself does not force the entire list (ArrM)" $ 
+    testError "Traverse' is too strict" id $ 
+    head $ head $ fst $ runIdentity $ steps (traverse' (arr id) :: Cell Identity [Int] [Int]) [[1, error "Bang !"]]
+  , testCase "traverse' by itself does not force the entire list (Cell)" $ 
+    testError "Traverse' is too strict" id $ 
+    head $ head $ fst $ runIdentity $ steps (traverse' cellId :: Cell Identity [Int] [Int]) [[1, error "Bang !"]]
+  , testCase "traverse' of a Cell composed with another Cell does not force the entire list" $ 
+    testError "Traverse' is too strict" id $ 
+    head $ head $ fst $ runIdentity $ steps (traverse' cellId >>> cellId  :: Cell Identity [Int] [Int]) [[1, error "Bang !"]]
```

Comment:
I think you can refactor the tests with `CellSimulation`, they will be easier to understand afterwards :)

Reply:
But here I'm not checking whether the cell produces a specific value, but just that it doesn't produce an error, so I don't think I can use `CellSimulation` for that, can I ?